opt_einsum/preprocessing.py

```diff
@@ -0,0 +1,244 @@
+import operator
+import functools
+import collections
+
+from .backends import get_func
+from .parser import find_output_str
+
+
+def unique(it):
```

Comment:
Can we type these? I cannot quite tell what this is doing, somewhat looks like a set operation.